src/sources/http.ts

```diff
@@ -47,14 +86,26 @@ export const createHTTPSource: SourceFactory = (options: any) => {
 
     let mtime
     const _lastModified = response.headers.get('last-modified')
+    const _etag = response.headers.get('etag')
+
     if (_lastModified) {
       mtime = new Date(_lastModified)
     }
 
+    let buffer
+
+    if (options.cache) {
+      const filename = join(options.cache, 'data', `${murmurhash(id)}`)
+      await fsp.mkdir(join(options.cache, 'data'), { recursive: true })
+      buffer = await response.buffer()
+      await fsp.writeFile(filename, buffer)
```

Comment:
We should also avoid using `fs` methods directly or consider about cache dir. storage layer should handle it based on used driver

Reply:
Would this mean creating an abstracted "Cache" engine with a default fs-based implementation? I did consider using unstorage, but didn't want it to have to stringify everthing

Reply:
Indeed we need binary support there. And that was why I had to pend ipx caching feature until supporting blob like data for unstorage... 

FYI until [ipx<=0.4.x](https://github.com/unjs/ipx/blob/v0.4.7/src/cache/fs.ts) we had fs based caching system for transformed assets but it was removed for serverless support where we don't have storage nor best place to storage cache data/meta.

Reply:
>  abstracted "Cache" engine

It would be abstracted storage engine + cache framework (cachify library to release) + input aware strategy (like etag checking)

Reply:
Moved task: https://github.com/unjs/unstorage/issues/21

Reply:
OK, so does this need to wait until unjs/unstorage#21 is done then?

Reply:
Yes

Reply:
Any movements on caching?